### PR TITLE
GLM alias

### PR DESCRIPTION
**Pull request by bhargavvader**
An attempt at #1768.
Would this do the trick? As for clearing it up with the docs; if the alias is in place, do we need to go and change all previous occurrences of it? 

It would also be a good idea to have the jupyter notebooks also added to Travis, so any failures in demos are caught easily.

### CODE REVIEW

**Comment by ColCarroll** [comment]
this won't work -- try updating `pymc3/glm/__init__.py` instead.  I just found out this pattern works, which I think will be useful:

```
In [1]: from collections import defaultdict, defaultdict as dd

In [2]: dd
Out[2]: collections.defaultdict

In [3]: defaultdict
Out[3]: collections.defaultdict
```

**Comment by ferrine** [comment]
Alias is named like a module, I think that's bad Idea. 
I wonder why we don't have a separate package with all models. We can name it like `pymc3.zoo` to avoid previous collision `model` and `models` and store glms in `pymc3.zoo.linear_models`

**Comment by twiecki** [comment]
@ferrine I agree, but just to add a deprecation warning. `zoo` is not a bad idea, @fonnesbeck?

**Comment by twiecki** [comment]
@ferrine And it would need to alias GLM.from_formula I think, right?

**Comment by twiecki** [comment]
@bhargavvader We also need to add a deprecation warning, as well as change all old occurrences.

**Comment by ferrine** [comment]
Additional alias is overhead in my opinion. We should have only `GLM` as a pep8 class name and `glm` as handy alias (using SHIFT is too annoing)

**Comment by twiecki** [comment]
@ferrine Yes, it's overhead. We'll just have it for a short while to give people a chance to switch and then drop it.

**Comment by ferrine** [comment]
Why? That's not that case when interface is too different so it will make some mess only.

**Comment by ferrine** [comment]
We can add a warning when wee see a string as a first argument:)

**Comment by twiecki** [comment]
@ferrine it's just a few lines:
```python
def glm(*args, **kwargs):
   warnings.warn('deprecated, use GLM.from_formula')
   return GLM.from_formula(*args, **kwargs)
```

**Comment by ferrine** [comment]
`glm = GLM` would be much better alias for long term. Now GLM accepts `x, y` as first arguments. If GLM and glm will have different interfaces it will be a real surprise when we then decide to unify them. 
I don't see any problem to change that small place of API in development branch. For production stable 3.0 can be used.

**Comment by twiecki** [comment]
How about we add a function that raises a Deprecation exception?

**Comment by ferrine** [comment]
That would be better. We should say that we'll add `glm = GLM` alias in the future (it's very handy)

**Comment by ColCarroll** [comment]
https://github.com/amueller/futurepast exists for deprecations -- it looks like sklearn isn't using it yet, but there [are three issues](https://github.com/scikit-learn/scikit-learn/search?q=futurepast&type=Issues&utf8=%E2%9C%93) where it is mentioned

**Comment by bhargavvader** [comment]
So what's the current decision for this? Is it deprecation warning + alias?
Also @ColCarroll , you mentioned that `glm = GLM` wouldn't work? Is it because the alias is named like a module?

**Comment by twiecki** [comment]
@bhargavvader First priority is to fix the occurrences of the old reference in the docs. Then we'll add a dummy `def glm(): raise DeprecationException` or something along these lines.

**Comment by ferrine** [comment]
I think that should be resolved as follows 

1. Create `pymc3.zoo` package 
2. Move `glm` module to `zoo.linear_models`
3. Create dummy `glm` function (raises exception) for a while
4. Fix docs
5. Later create a shortcut `glm = GLM`

Any thoughts on this?

**Comment by bhargavvader** [comment]
That'd require a fair amount of restructuring, right? I'll wait for a little more discussion before making any changes - but I do think that there should some sort of an immediate solution (at least in the form of doc changing and alias).

**Comment by ferrine** [comment]
What's the status of this PR?

**Comment by bhargavvader** [comment]
Still waiting for more feedback. @twiecki , what do you think?
Also, about making changes in the docs - where all would there a need to be changes? In the website and notebooks?

**Comment by twiecki** [comment]
I think we should leave this as is for now. Perhaps at some point we should have an external repository for a model zoo. Thus, lets raise a DeprecationException for the compat wrapper and fix the docs here.

**Comment by twiecki** [comment]
@bhargavvader Sound good?

**Comment by bhargavvader** [comment]
Yeah - I've been caught up with some work at school, will wrap up this over the weekend/next week.

**Comment by bhargavvader** [comment]
@twiecki, has the `plot_posterior_predictive` method been removed or moved somewhere else? I can't find it in `glm` .

**Comment by bhargavvader** [comment]
Also: I am making changes in the notebooks, how will these be reflected in the docs/webpages? Does it have to be done manually or does it 'sync' up on it's own? I'm pretty unaware with the details of linking notebooks with webpages.

**Comment by twiecki** [comment]
it's possible that it got removed inadvertently.

**Comment by ColCarroll** [comment]
Looks like here: https://github.com/pymc-devs/pymc3/commit/d37e7a12c8d1ce1de0e17de8f18842d3014dc812

Might be a good opportunity to add it to the new `plots` module!

**Comment by bhargavvader** [comment]
Yeah, sounds good - I'll open a separate PR for it, wait for it to get merged and then continue with re-doing the docs - I'd need that method to run the notebooks. Does that sound ok @twiecki ?

**Comment by twiecki** [comment]
Yes, perfect. Also, don't assume that function to be well written, I'm sure it could be done better.

**Comment by twiecki** [comment]
`pymc3/plots/posteriorplot.py:120: [W0404(reimported), plot_posterior_predictive_glm] Reimport 'matplotlib.pyplot' (imported line 2)`

**Comment by bhargavvader** [comment]
Whoops - removed the extra import!

File: pymc3/glm/linear.py [hunk 1]
```diff
@@ -134,3 +134,5 @@ def from_formula(cls, formula, data, priors=None,
         labels = x.design_info.column_names
         return cls(np.asarray(x), np.asarray(y)[:, 0], intercept=False, labels=labels,
                    priors=priors, vars=vars, family=family, name=name, model=model)
+
+glm = GLM
```
Comment (bhargavvader) [inline pymc3/glm/linear.py:138]: I should remove the alias, right?

Comment (twiecki) [reply pymc3/glm/linear.py:138]: I think the alias is good for backwards compat.

**Comment by twiecki** [comment]
Thanks @bhargavvader!

**Pull request merged**